src/bid_order.rs

```diff
@@ -44,7 +44,9 @@ pub struct BidOrderV1 {
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct BidOrderV2 {
```

Comment:
I didnt see this earlier, but if you want to be able to perform a migration of the old bid orders (if any exist), you need to create a new struct, `BidOrderV3` with the `accumulated_X` fields

then in the `migrate()` entry point, you need to do this flow:
1. load in the `BidOrderV2` formatted orders from the contract storage (can delete)
2. convert each `BidOrderV2` into a `BidOrderV3` struct
3. write each new `BidOrderV3` order to contract storage

To allow for this backwards compatibility, you need to keep the previous version of the struct, `BidOrderV2` in the contract, and then add the new version `BidOrderV3`

this is basically why there are so many old versions. Each one allows for migrations

Reply:
@brashidnia-figure Yes, I started on this path after we talked, which prompted me to open #111 (I figured it was better to clean up the old structs first before adding a new one)